pkg/networkservice/common/timeout/server.go

```diff
@@ -21,80 +23,170 @@ import (
 	"context"
 	"time"
 
+	"github.com/edwarnicke/serialize"
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/empty"
-	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/pkg/errors"
 
-	"github.com/edwarnicke/serialize"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
-	"github.com/networkservicemesh/sdk/pkg/networkservice/core/trace"
-	"github.com/networkservicemesh/sdk/pkg/tools/extend"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
 )
 
 type timeoutServer struct {
-	onTimeout   *networkservice.NetworkServiceServer
-	connections map[string]*time.Timer
-	executor    serialize.Executor
+	ctx       context.Context
+	timers    timerMap
+	executors executorMap
+}
+
+type timer struct {
```

Comment:
Please cover the case when stopCh is using or let's just delete this struct.

Reply:
I've reworked it with such thing:
```go
timerPtr := new(*time.Timer)
*timerPtr = time.AfterFunc(time.Until(expireTime), func() {
	executor.AsyncExec(func() {
		if timer, ok := t.timers.Load(conn.GetId()); !ok || timer != *timerPtr {
			logEntry.Warnf("timer has been already stopped: %v", conn.GetId())
			return
		}
		if err := t.close(t.ctx, conn, next.Server(ctx)); err != nil {
			logEntry.Errorf("failed to close timed out connection: %v %+v", conn.GetId(), err)
		}
	})
})
```
so there is no more need for the `stopCh`.